### ISSUE TITLE
Replace autotools build system

### ISSUE THREAD

**Issue by ChristianBeer** [BOINC/boinc#1882]
The current autotools centered build system is very fragile and hard to maintain (if you are not a autotools expert). It is also Unix/Linux centered and only half heartedly supports cross compilation (e.g. mingw).

I propose to replace it gradually with a [cmake](https://cmake.org/) based build system which offers multi-platform and multi-compiler support and an easier language to write build configuration files. It can for example build Visual Studio project files as well as regular Makefiles as well as control files for other build tools.

I already did some initial testing and can create config.h to replace autoheader and have some simple rules to build basic libraries. The challenging part is backwards compatibility (what special checks are still relevant and what can be omitted) and creating/using static libraries for release builds which I have not yet explored.

**Comment by AenBleidd** [comment]
@ChristianBeer Agreed. I have some experience creating cmake scripts so I can try to help you with this.

**Comment by JamesMolson** [comment]
I am working on this right now and should have an early (i.e. there might be some missing features) pull request ready for people to view within the next few days.

**Comment by talregev** [comment]
I want to start a process to convert boinc compilation to CMake.
@ChristianBeer @JamesMolson Do you want to join the efforts?

**Comment by BryanQuigley** [comment]
I've been wrestling with autotools repeatedly.  Any chance to help with this?

**Comment by talregev** [comment]
@BryanQuigley What help do you need? 
Also we compile boinc with autotools inside the ci. You can see an example how to compile in your system.

**Comment by AenBleidd** [comment]
I do agree that autotools centered build system is very complex, however it's proved to work for two decades already.
Yes. it's used at the moment on *nix systems only (including Android) but it works fine and stable.
Moving it to cmake will require a huge effort and a log of resources (and I even not sure how to test it afterwards on all the systems and configuration we currently support running/building BOINC on).
Whine in general this sounds interesting, this is not something we are gonna work.
Maybe one day we will reconsider this, but until then I don't see any reason keeping this ticket open.
